scripts/utility.py

```diff
@@ -1665,6 +1665,43 @@ def change_relationship_values(
                     if log_text not in rel.log:
                         rel.log.append(log_text)
 
+def get_kin_groups(cat, include_dead=True):
+    """
+    Retrieves the kin groups (close kin, kin, and distant kin) for the cat.
+    :param include_dead: Whether to include dead cats in the kin groups (default: True).
+    :return: A tuple containing three dictionaries: close_kin, kin, and distant_kin.
+             - close_kin: Dictionary of close kin groups (parents, siblings, children, mates, former mates).
+             - kin: Dictionary of kin groups (grandparents, aunts/uncles, cousins, grandkits).
+             - distant_kin: Dictionary of distant kin groups.
+    """
+    close_kin = {
+        "gen_parents": [cat.all_cats.get(cat_id) for cat_id in cat.get_parents()],
+        "gen_siblings": [cat.all_cats.get(cat_id) for cat_id in cat.get_siblings()],
+        "gen_children": [cat.all_cats.get(cat_id) for cat_id in cat.get_children()],
+        "gen_mates": [cat.all_cats.get(cat_id) for cat_id in cat.mate],
+        "gen_former_mates": [cat.all_cats.get(cat_id) for cat_id in cat.previous_mates]
+    }
+
+    kin = {
+        "gen_grandparents": [cat.all_cats.get(cat_id) for cat_id in cat.all_cats if cat.all_cats.get(cat_id).is_grandparent(cat)],
```

Comment:
The "get_kin_groups" function is not very performant. To improve it at least a little add the new "kin" to the inheritance class. It seems it was done but not used?